File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        total = 0
        for i in range(len(mat)): # 0 1 2 3 
            for j in range(len(mat)): # 0 1 2 3 
                if i == j: # (0 0) (1 1) (2 2) (3 3) 
                    total += mat[i][j] 
            total += mat[i][len(mat)-i-1]
        if len(mat) % 2 == 1:
            total -= mat[len(mat)//2][len(mat)//2]
        return total

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int total = 0;
        int n = mat.length;
        for (int i = 0; i < n;i++) {
            total += mat[i][i];
            total += mat[i][n - i - 1];
        }
        return n % 2 == 0 ? total : total - mat[(n-1)/2][(n-1)/2];    
    }
}